A part of Sprint 3.geo A for voxel

oyo-rdc opened the pull request:
## File

Containing voxel

oyo-rdc committed:
A part of Sprint 3.geo A for voxel

oyo-rdc committed:
Merge branch 'bSI-InfraRoom:main' into main

oyo-rdc committed:
part of Sprint 3.geo A for voxel

oyo-rdc committed:
Add files via upload

oyo-rdc commented:
Updated the file

larswik commented:
@pjanck : Seems like a bug in the checker (object reference not set to an instance of an object).

oyo-rdc committed:
Add files via upload

Commented out geophysical survey common property set

SergejMuhic commented:
Alright, checked the file and there is a bunch of things wrong with it. The checker expects a certain completeness and since it is not there, we get a null value.

The following should be corrected first as these mistakes are already very deep on schema level:
StepSerializer: #27 IfcLocalPlacement attribute PlacementRelTo of type IfcObjectPlacement does not match provided type IfcAxis2Placement3D.
StepSerializer: #1002 IfcLocalPlacement attribute PlacementRelTo of type IfcObjectPlacement does not match provided type IfcAxis2Placement3D.
StepSerializer: #1002 IfcLocalPlacement attribute RelativePlacement of type IfcAxis2Placement does not match provided type IfcLocalPlacement.
StepSerializer: #1012 IfcDerivedUnit attribute Name is required but does not have a value.

The `PlacementRelTo` attribute should not point to `IfcAxis2Placement` but to another `IfcObjectPlacement`. On the other hand, the attribute `RelativePlacement` should point to an `IfcLocalPlacement`. Apparently the two attributes were switched. `PlacementRelTo` should be on position 1 and `RelativePlacement` on position 2. This is the case for # 1002.

The last issue for `IfcDerivedUnit` is not that critical but `Name` is mandatory.

oyo-rdc committed:
Add files via upload

oyo-rdc commented:
@SergejMuhic 
I uploaded the revised file and failed the checker again. Would you please provide where it is wrong?
Also, about IfcDerivedUnit, the attribute "Name" seems optional in the IFC4x3 add2 documentation (https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/lexical/IfcDerivedUnit.htm), but it seems mandatory in the 4x4 draft documentation. Has it been changed?

SergejMuhic commented:
> @SergejMuhic I uploaded the revised file and failed the checker again. Would you please provide where it is wrong?

We are still dealing with serialization errors on voxel (see inline for further explanation):
StepSerializer: #108 IfcShapeRepresentation attribute ContextOfItems of type IfcRepresentationContext does not match provided type IfcLocalPlacement. -> `IfcShapeRepresentation`.`ContextOfItems` (first attribute) points to 103 `IfcLocalPlacement`. Should point to an `IfcRepresentationContext`.
StepSerializer: Object of type 'THC.IFC4X4.IfcRepresentationResource.IfcGeometricRepresentationSubContext' cannot be converted to type 'THC.IFC4X4.IfcGeometryResource.IfcRepresentationItem'.for elements: (#106) -> `IfcShapeRepresentation`.`Items` points to `IfcGeometricRepresentationContext`, should be pointing to `IfcRepresentationItem`
StepSerializer: #1007 IfcShapeRepresentation attribute ContextOfItems of type IfcRepresentationContext does not match provided type IfcLocalPlacement. -> same error as the first but for 1007.

> Also, about IfcDerivedUnit, the attribute "Name" seems optional in the IFC4x3 add2 documentation (https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/lexical/IfcDerivedUnit.htm), but it seems mandatory in the 4x4 draft documentation. Has it been changed?

No, you are right. This will be corrected in the reconciliation efforts. This issue does not influence a drop out of the check procedure however, so just ignore for now.

oyo-rdc committed:
Add files via upload

oyo-rdc commented:
@SergejMuhic 
Thank you for checking. Clearly my mistakes -- I forgot to update the line numbers. Manual editing is not easy...
The corrected file still fails the check. Is there a way I can see error messages from the checker?

SergejMuhic commented:
> @SergejMuhic Thank you for checking. Clearly my mistakes -- I forgot to update the line numbers. Manual editing is not easy... The corrected file still fails the check. Is there a way I can see error messages from the checker?

Two issues:
- in borehole: 
```IFCLABEL('\X2\76db571f0028792b6df73058308a78028cea7c98571f0029\0\ Fill (Sandy clay with gravel)')```
the closing encoding is ```\0\``` and should be ```\X0\```
- in voxel:
```
#1015= IFCPROPERTYENUMERATEDVALUE('SurveyMethod',$,(.PASSIVE_SEISMIC.),$);
```
should be
```
#1015= IFCPROPERTYENUMERATEDVALUE('SurveyMethod',$,IFCLABEL(.PASSIVE_SEISMIC.),$);
```

oyo-rdc committed:
Add files via upload

oyo-rdc committed:
Update schema-check.yml

oyo-rdc committed:
Add files via upload

oyo-rdc committed:
Add files via upload

oyo-rdc committed:
Add files via upload

oyo-rdc committed:
Add files via upload

oyo-rdc committed:
Add files via upload

oyo-rdc committed:
Merge branch 'bSI-InfraRoom:main' into main

oyo-rdc committed:
Updated borehole file

Updated borehole.ifc file. Added borehole.XML, which is the source of the data in borehole.ifc, and borehole.pdf, which is a borehole log generated based on the xml file.

oyo-rdc committed:
Added the PDF of the borehole log in (partly) Engsh

oyo-rdc committed:
Add files via upload

Corrected errors that Stefan pointed out

oyo-rdc committed:
Uploaded a borehole with checmical lab test

oyo-rdc committed:
Corrected Borehole_ChemicalLabTest and voxel

pjanck commented on files/oyo/Borehole_ChemicalLabTest.ifc:
```diff
@@ -0,0 +1,1241 @@
+HEADER;
+FILE_DESCRIPTION(('ViewDefinition [ReferenceView_V1.2]'),'2;1');
+FILE_NAME('Borehole_ChemicalLabTest.ifc','2024-01-01T00:00::00',('Kazunori Takahashi'),('OYO Corporation'),'','','');
+FILE_SCHEMA(('IFC4X3'));
```
```suggestion
FILE_SCHEMA(('IFC4X4'));

```

For the checker to pick it up :)

oyo-rdc committed:
Update Borehole_ChemicalLabTest.ifc

oyo-rdc committed:
Corrected erros in borehole.ifc

oyo-rdc committed:
Corrections made in borehole.ifc

oyo-rdc committed:
Add files via upload

oyo-rdc commented:
Uploaded a new ifc file (tokyo/tokyo-sections.ifc) that contains a geological section. Currently, it uses IFC4x3 entities (i.e. building element proxy etc.) to view in BIM vision because the file is not rendered correctly in ifcviewer (please see screenshots below). @peterrdf Would be appreciated if you could check the file and look into why this happens. Thank you.
![image](https://github.com/bSI-InfraRoom/IFC-Tunnel-Deployment/assets/127815423/a05cf066-ac37-4e44-89be-7266aae83d7a)
![image](https://github.com/bSI-InfraRoom/IFC-Tunnel-Deployment/assets/127815423/dcca4385-d95b-49c7-9dcb-f19f2729d94b)

SergejMuhic commented:
Alright, the file is good. @peterrdf found a small bug. Here the screenshot:
![image](https://github.com/user-attachments/assets/371fcf02-1fbf-4512-9624-87022e55bc50)

peterrdf commented:
my bad indeed, the bug is fixed, viewer and library will be updated early next week (https://[rdf.bg/product-list/ifc-engine/ifc-viewer/](https://rdf.bg/product-list/ifc-engine/ifc-viewer/))

oyo-rdc commented:
Thank you, @peterrdf, @SergejMuhic for checking. I'll test our files with the updated version.
